src/source.rs

```diff
@@ -276,6 +278,55 @@ impl ImageLocation for StreamLocation {
     }
 }
 
+/// Subcommand to list objects available in stream metadata.
+pub fn list_stream(config: &ListStreamConfig) -> Result<()> {
+    // fetch stream metadata
+    let stream_url = build_stream_url(&config.stream, config.stream_base_url.as_ref())?;
+    let client = reqwest::Client::new();
+    let stream = fetch_stream(client, &stream_url)?;
+
+    // build results
+    // default to arch list
+    let mut header = format!("Architectures for stream {}:", &config.stream);
+    let mut result = Vec::from_iter(stream.architectures.keys());
+    if let Some(architecture_name) = config.architecture.as_ref() {
+        // the user specified an arch
+        if let Some(architecture) = stream.architectures.get(architecture_name) {
+            // ...and we found it
+            header = format!(
+                "Platforms for stream {}, architecture {}:",
+                &config.stream, architecture_name
+            );
+            result = Vec::from_iter(architecture.artifacts.keys());
+            if let Some(platform_name) = config.platform.as_ref() {
+                // the user also specified a platform
+                if let Some(platform) = architecture.artifacts.get(platform_name) {
+                    // ...and we found it
+                    header = format!(
+                        "Formats for stream {}, architecture {}, platform {}:",
```

Comment:
Does "formats" mean media types here?

Reply:
"Formats" in the sense of [stream metadata](https://github.com/coreos/fedora-coreos-tracker/blob/master/metadata/stream/rationale.yaml).  They don't exactly map to content-types, since we have formats like `iso` and `pxe`.